Plugin scaffold CLI

mostafa opened the pull request:
This PR updates the docs with the `gatewayd plugin scaffold` command and also updated the plugin developers' guide.

FYI: @zeina1i 

## Changed docs
- https://docs-git-plugin-scaffold-cli-gatewaydio.vercel.app/using-gatewayd/CLI/
- https://docs-git-plugin-scaffold-cli-gatewaydio.vercel.app/developing-plugins/plugin-developers-guide/

## Related PR
- https://github.com/gatewayd-io/gatewayd/pull/518

zeina1i commented on developing-plugins/plugin-developers-guide.md:
```diff
@@ -90,22 +103,22 @@ You can test your plugin locally by running GatewayD CLI in development mode. Th
+## Step 6: Publish your plugin to GitHub using this `release.yaml` workflow and this `Makefile`
 
-If you want to publish your plugin to GitHub, you can use the following workflow and `Makefile` to release your plugin. Copy the [`release.yaml`](https://github.com/gatewayd-io/gatewayd-plugin-cache/blob/main/.github/workflows/release.yaml) workflow and this [`Makefile`](https://github.com/gatewayd-io/gatewayd-plugin-cache/blob/main/Makefile) into the `.github/workflows/` and the root directory o ...
+If you want to publish your plugin to GitHub, you can use the following workflow and `Makefile` to release your plugin. Copy the [`release.yaml`](https://github.com/gatewayd-io/gatewayd/blob/main/plugin/.template/project/%7B%7B%20plugin_name%20%7D%7D/.github/workflows/release.yaml) workflow and this [`Makefile`](https://github.com/gatewayd-io/gatewayd/blob/main/plugin/.template/project/%7B%7B%20p ...
```
considering these files are generated, there is no need to copy them. However I think, this sentence is remained from previous doc :)

zeina1i commented on developing-plugins/plugin-developers-guide.md:
```diff
@@ -90,22 +103,22 @@ You can test your plugin locally by running GatewayD CLI in development mode. Th
+If you want to publish your plugin to GitHub, you can use the following workflow and `Makefile` to release your plugin. Copy the [`release.yaml`](https://github.com/gatewayd-io/gatewayd/blob/main/plugin/.template/project/%7B%7B%20plugin_name%20%7D%7D/.github/workflows/release.yaml) workflow and this [`Makefile`](https://github.com/gatewayd-io/gatewayd/blob/main/plugin/.template/project/%7B%7B%20p ...
 
 {: .note }
 > You must modify the example `Makefile`, `release.yaml` and `test.yaml` files to match your plugin.
```
Makefile and release.yaml don't need modification.
So these two files can be removed from sentence.

zeina1i reviewed (commented):
Thanks @mostafa 🙏

It looks good to me,
I left a few comments, let's discuss when you have a chance.

mostafa commented on developing-plugins/plugin-developers-guide.md:
```diff
@@ -15,30 +15,30 @@ The usage of plugin from the user perspective is described [here](/using-plugins
-{: .note }
-> You can also use the [GatewayD plugin template for Python](https://github.com/gatewayd-io/plugin-template-python) repository to create a new repository for your plugin. The project is not as mature as the Go template and might have some rough edges.
+1. Generate a plugin scaffold using the `gatewayd plugin scaffold` command.
+2. Update the `gatewayd_plugins.yml` file with the correct information.
```
Since the user is developing the plugin, they can just disable checksum verification, hence using `./gatewayd run --dev` or `make run`. I don't see why we should automate this at this point.

zeina1i replied:
I will create the ticket for automation, after review 🤞

zeina1i replied:
Ticket is created:
https://github.com/gatewayd-io/gatewayd/issues/529

mostafa commented on developing-plugins/plugin-developers-guide.md:
````diff
@@ -53,11 +53,24 @@ This is the structure of the template:
+After you have created the `input.yaml` file, run the following command to generate the plugin scaffold:
+
+```bash
+gatewayd plugin scaffold --input-file input.yaml --output-dir gatewayd-plugin-test
````
This is conformant to the `plugin install` command that uses the same flag for choosing the directory to install plugins into. If you think this is not a good flag name, create a ticket and we'll change in all places at once.

mostafa commented on developing-plugins/plugin-developers-guide.md:
```diff
@@ -71,12 +84,12 @@ The following fields are optional:
 - `args`: The arguments that are passed to the plugin.
 - `checksum`: The checksum of the plugin binary.
```
See above.

mostafa commented:
@zeina1i Addressed your comments in https://github.com/gatewayd-io/docs/pull/51/commits/25268e355497296d96a7c2a3bea2ae7661988f60 and left some comments as well.

zeina1i reviewed (approved):
Thanks for new changes @mostafa 🙏
Approved 🚀

mostafa commented:
@zeina1i I'll merge it after releasing a new version of GatewayD.
